R/backend-mssql.R

```diff
@@ -250,7 +250,7 @@ simulate_mssql <- function(version = "15.0") {
 
       `[` = function(x, i) {
         i <- with_mssql_bool(i)
-        build_sql("CASE WHEN (", i, ") THEN (", x, ") END")
+        glue_sql2("CASE WHEN ({i}) THEN ({x}) END", .con = sql_current_con())
```

Comment:
I considered defaulting to `sql_current_con()` but I was against it for a couple of reasons:

* defaulting later to `sql_current_con()` felt much less work than the other way round
* it is a bit magic and can easily lead to subtle bugs
* I'd rather not do it in case we decide to later export `glue_sql2()`

But having it as the first argument sounds reasonable. Especially, because you rarely ever would like to pipe into it.